### PR TITLE
Updated doc structure and added contribution guide

### PR DESCRIPTION
**Pull request by wklumpen**
This pull requests largely consolidates the documentation available on Read the Docs based on Issue https://github.com/transit-analytics-lab/spur/issues/46. 

Before this is changed to a full pull request, the following must be done:

- [x] Creation of a "contribution guide" page showing how to set up a development environment for Spur
- [x] Update of the structure of the "User Guide" and the developer reference
- [x] Create a "quick start" example vignette using Line 4

### CODE REVIEW

File: docs/source/index.rst [hunk 1]
```diff
@@ -3,38 +3,37 @@
...
+The model balances simplification with flexibility, and requires substantially 
+less data and expertise than traditional microscopic simulation models.
 
-This documentation website is designed both as a user manual and as a technical
+Spur is currently **under initial development**. This means that while features
+are available and can in principle be used for simulation purposes, the design of the
+simulation and codebase are likely to change rapidly without much thought for backwards
+compatability or overall impacts. 
+
+This documentation serves as a development reference, technical documentation source, and
```
Comment (peterlai1) [inline docs/source/index.rst:21]: > This documentation serves as a development reference, technical documentation source, and eventual user guide for the software platform. 
> 
> The documentation here is designed both as a user manual and as a technical documentation source for developers wishing to adapt or extend the software or build their own plugins.

Are both of these paragraphs meant to be included, or just one of them?

**Review by peterlai1 — COMMENTED** [review]
Looks great! Corrected a couple of very minor typos and left a comment about a potentially duplicate paragraph.

**Comment by wklumpen** [comment]
@peterlai1 take a look.

**Review by peterlai1 — APPROVED** [review]
All good! Thanks!

**Pull request merged**